openapi-validator/src/main/java/io/ballerina/openapi/validator/HeaderValidator.java

```diff
@@ -0,0 +1,181 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.openapi.validator;
+
+import io.ballerina.compiler.syntax.tree.AnnotationNode;
+import io.ballerina.compiler.syntax.tree.DefaultableParameterNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeList;
+import io.ballerina.compiler.syntax.tree.RequiredParameterNode;
+import io.ballerina.openapi.validator.error.CompilationError;
+import io.ballerina.openapi.validator.model.MetaData;
+import io.swagger.v3.oas.models.media.ArraySchema;
+import io.swagger.v3.oas.models.media.Schema;
+import io.swagger.v3.oas.models.parameters.HeaderParameter;
+import io.swagger.v3.oas.models.parameters.Parameter;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static io.ballerina.openapi.validator.Constants.HEADER_NAME;
+import static io.ballerina.openapi.validator.Constants.HTTP_HEADER;
+import static io.ballerina.openapi.validator.Constants.SQUARE_BRACKETS;
+import static io.ballerina.openapi.validator.TypeValidatorUtils.convertOpenAPITypeToBallerina;
+import static io.ballerina.openapi.validator.ValidatorUtils.extractAnnotationFieldDetails;
+import static io.ballerina.openapi.validator.ValidatorUtils.extractReferenceType;
+import static io.ballerina.openapi.validator.ValidatorUtils.getNormalizedPath;
+import static io.ballerina.openapi.validator.ValidatorUtils.getNumberFormatType;
+import static io.ballerina.openapi.validator.ValidatorUtils.reportDiagnostic;
+import static io.ballerina.openapi.validator.ValidatorUtils.unescapeIdentifier;
+
+/**
+ * This contains the header validation.
+ *
+ * @since 1.1.0
+ */
+public class HeaderValidator extends AbstractMetaData implements SectionValidator, Validator {
+    private final Map<String, Node> balHeaders;
+    private final List<Parameter> oasParameters;
+
+    public HeaderValidator(MetaData metaData, Map<String, Node> balHeaders, List<Parameter> oasParameters) {
+        super(metaData.getContext(), metaData.getOpenAPI(), metaData.getPath(), metaData.getMethod(),
+                metaData.getSeverity(), metaData.getLocation());
+        this.balHeaders = balHeaders;
+        this.oasParameters = oasParameters;
+    }
+
+    @Override
+    public void validate() {
+        //Ballerina to openAPI header validation.
+        validateBallerina();
+        //OAS->Ballerina header validation
+        if (oasParameters == null) {
+            return;
+        }
+        validateOpenAPI();
+    }
+
+
+    /**
+     * This function is used to validate the ballerina resource header against to openapi header.
+     *
```

Comment:
Additional line

Reply:
I will remove it, I thought we need to have an extra line after Javadoc. 🙂